### PR TITLE
11790_home_page

### PR DESCRIPTION
**Pull request by tuananhp-1844**
https://edu-redmine.sun-asterisk.vn/issues/11790

### CODE REVIEW

File: app/Http/Controllers/Questions/QuestionController.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Repositories\Contracts\QuestionRepositoryInterface;
+use App\Repositories\Contracts\CategoryRepositoryInterface;
+use App\Http\Requests\Questions\CreateQuestionRequest;
+
+class QuestionController extends Controller
+{
+    private $questionRepository;
+    private $categoryRepository;
```
Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.Files.LineLength.TooLong</br>**Severity**: warning</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 16</br>Line exceeds 120 characters; contains 131 characters

Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 16</br>Opening brace should be on a new line

File: app/Http/Requests/Languages/LanguageRequest.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
```
Comment (framgiaciv3) [inline app/Http/Requests/Languages/LanguageRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/Languages/LanguageRequest.php</br>**Line**: 27</br>Each line in an array declaration must end in a comma

File: app/Http/Requests/Questions/CreateQuestionRequest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return Auth::check();
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
```
Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 28</br>String "category" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 28</br>String "required" does not require double quotes; use single quotes instead

File: app/Http/Requests/Questions/CreateQuestionRequest.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        return [
```
Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 29</br>String "title" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 29</br>String "required" does not require double quotes; use single quotes instead

File: app/Http/Requests/Questions/CreateQuestionRequest.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            "category" => "required",
```
Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 30</br>String "content" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 30</br>String "required" does not require double quotes; use single quotes instead

File: app/Http/Requests/Questions/CreateQuestionRequest.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            "title" => "required",
```
Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 31</br>String "ask" does not require double quotes; use single quotes instead

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 31</br>Each line in an array declaration must end in a comma

Comment (framgiaciv3) [inline app/Http/Requests/Questions/CreateQuestionRequest.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Requests/Questions/CreateQuestionRequest.php</br>**Line**: 31</br>String "array" does not require double quotes; use single quotes instead

File: app/Providers/AppServiceProvider.php [hunk 7]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
      */
     public function register()
     {
-        //
+        $models = [
+            'Question',
+            'Category'
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 19</br>Comma required after last value in array declaration

File: app/Providers/AppServiceProvider.php [hunk 8]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+        ];
+        foreach ($models as $key => $model) {
+            $this->app->bind(
+				"App\\Repositories\\Contracts\\{$model}RepositoryInterface",
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 23</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 9]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+				"App\\Repositories\\Eloquents\\{$model}Repository"
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 24</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 10]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+			);
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 25</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Http/Controllers/Questions/QuestionController.php (hunk 1 above)
Comment (framgiaciv3) [inline app/Http/Controllers/Questions/QuestionController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.Functions.MultiLineFunctionDeclaration.BraceOnSameLine</br>**Severity**: error</br>**File**: app/Http/Controllers/Questions/QuestionController.php</br>**Line**: 16</br>Opening brace should be on a new line

File: app/Providers/AppServiceProvider.php [hunk 11]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
     public function register()
     {
-        //
+        $models = [
+            'Question',
+            'Category',
+        ];
+        foreach ($models as $key => $model) {
+            $this->app->bind(
+				'App\\Repositories\\Contracts\\{$model}RepositoryInterface',
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 23</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 12]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+				'App\\Repositories\\Eloquents\\{$model}Repository',
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 24</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 13]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+			);
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 25</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+use App\Repositories\Contracts\BaseRepositoryInterface;
+use Illuminate\Container\Container as App;
+use Illuminate\Database\Eloquent\Model;
+
+abstract class BaseRepository implements BaseRepositoryInterface
+{
+    /** var
+     */
+    protected $model;
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 15</br>Whitespace found at end of line

File: app/Repositories/Eloquents/BaseRepository.php [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+    public function __construct()
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 17</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+use Illuminate\Database\Eloquent\Model;
+
+abstract class BaseRepository implements BaseRepositoryInterface
+{
+    /** var
+     */
+    protected $model;
+
+    public function __construct() 
+    {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 18</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+		$this->setModel();
+	}
+
+    abstract public function getModel();
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 22</br>Whitespace found at end of line

File: app/Repositories/Eloquents/BaseRepository.php [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+
+    public function setModel()
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 24</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+
+    public function __construct() 
+    {
+		$this->setModel();
+	}
+
+    abstract public function getModel();
+
+    public function setModel() 
+    {
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 25</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+		$this->model = app()->make(
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 26</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Repositories/Eloquents/BaseRepository.php [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+			$this->getModel()
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/BaseRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/BaseRepository.php</br>**Line**: 27</br>Spaces must be used to indent lines; tabs are not allowed

File: app/Providers/AppServiceProvider.php [hunk 22]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
     public function register()
     {
-        //
+        $models = [
+            'Question',
+            'Category',
+        ];
+        foreach ($models as $key => $model) {
+            $this->app->bind(
+			 'App\\Repositories\\Contracts\\{$model}RepositoryInterface',
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 23</br>Spaces must be used to indent lines; tabs are not allowed

Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 23</br>Multi-line function call not indented correctly; expected 16 spaces but found 13

File: app/Providers/AppServiceProvider.php [hunk 23]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+			 'App\\Repositories\\Eloquents\\{$model}Repository',
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 24</br>Spaces must be used to indent lines; tabs are not allowed

Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 24</br>Multi-line function call not indented correctly; expected 16 spaces but found 13

File: app/Providers/AppServiceProvider.php [hunk 24]
```diff
@@ -14,7 +14,16 @@ class AppServiceProvider extends ServiceProvider
...
+			);
```
Comment (framgiaciv3) [inline app/Providers/AppServiceProvider.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Generic.WhiteSpace.DisallowTabIndent.TabsUsed</br>**Severity**: error</br>**File**: app/Providers/AppServiceProvider.php</br>**Line**: 25</br>Spaces must be used to indent lines; tabs are not allowed

**Comment by tuananhp-1844** [comment]
ready

File: .bowerrc [hunk 25]
```diff
@@ -0,0 +1,3 @@
+{
+    "directory": "public/bower_components/"
+}
```
Comment (BunnyPi04) [inline .bowerrc]: thiếu dòng trắng cuối file

File: resources/views/layouts/app.blade.php [hunk 26]
```diff
@@ -0,0 +1,80 @@
...
+    <meta name="csrf-token" content="{{ csrf_token() }}">
+
+    <title>{{ config('app.name', 'Laravel') }}</title>
+
+    <!-- Scripts -->
+    <script src="{{ asset('js/app.js') }}" defer></script>
+
+    <!-- Fonts -->
+    <link rel="dns-prefetch" href="//fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css?family=Nunito" rel="stylesheet" type="text/css">
```
Comment (BunnyPi04) [inline resources/views/layouts/app.blade.php]: tải font về nhé

File: resources/views/layouts/app.blade.php [hunk 27]
```diff
@@ -0,0 +1,80 @@
...
+                                </a>
+
+                                <div class="dropdown-menu dropdown-menu-right" aria-labelledby="navbarDropdown">
+                                    <a class="dropdown-item" href="{{ route('logout') }}"
+                                       onclick="event.preventDefault();
+                                                     document.getElementById('logout-form').submit();">
+                                        {{ __('Logout') }}
+                                    </a>
+
+                                    <form id="logout-form" action="{{ route('logout') }}" method="POST" style="display: none;">
```
Comment (BunnyPi04) [inline resources/views/layouts/app.blade.php]: style viết vào file css

**Comment by tuananhp-1844** [comment]
ready

File: resources/views/layouts/app.blade.php [hunk 28]
```diff
@@ -0,0 +1,76 @@
...
+<script src="{{ asset('bower_components/asset-project1-sun/js/jquery.bxslider.min.js') }}"></script>
+<script src="{{ asset('bower_components/simplemde/dist/simplemde.min.js') }}"></script>
+{{-- <script src="{{ asset('bower_components/dropzone/dist/dropzone.js') }}"></script> --}}
+{{-- <script src="{{ asset('bower_components/taggle/src/taggle.js') }}"></script> --}}
+<script src="{{ asset('js/custom.js') }}"></script>
+<!-- End js -->
+
+</body>
+
+</html>
```
Comment (BunnyPi04) [inline resources/views/layouts/app.blade.php]: thiếu dòng trắng cuối file

**Comment by tuananhp-1844** [comment]
ready

**Pull request merged**